### PR TITLE
Swift SDK v2

### PR DESCRIPTION
**Pull request by kapitoshka438**

### CODE REVIEW

**Review by mplooster — APPROVED** [review]
Approved.

As a side note, do you like the names given to some of the autogenerated classes? Some of them seem very unwieldy, like `Cobaltspeech_Diatheke_SessionOutput`. We might be able to change that by modifying the [package name](https://github.com/cobaltspeech/sdk-diatheke/blob/master/grpc/diatheke.proto#L17) in the protobuf file. Although, such a change would probably be reviewed by other developers first.

**Comment by kapitoshka438** [comment]
> Approved.
> 
> As a side note, do you like the names given to some of the autogenerated classes? Some of them seem very unwieldy, like `Cobaltspeech_Diatheke_SessionOutput`. We might be able to change that by modifying the [package name](https://github.com/cobaltspeech/sdk-diatheke/blob/master/grpc/diatheke.proto#L17) in the protobuf file. Although, such a change would probably be reviewed by other developers first.

Is there a way to avoid using `Cobaltspeech_Diatheke_` in classes and other types names? Just leave them as `SessionOutput`, `SessionInput`, `TokenData` etc.? As for the package name, I manually set the name of the library as `Diatheke`. So, when a developer imports the package, it includes it with `import Diatheke` and access the classes like `Cobaltspeech_Diatheke_SessionOutput`. It is fine to shrink it to `SessionOutput`. Even if a project already has a type with this name, they can avoid conflicts specifying the package/namespace name like `Diatheke.SessionOutput`.

File: grpc/swift-diatheke/Client.swift [hunk 1]
```diff
@@ -66,185 +66,203 @@ public extension Cobaltspeech_Diatheke_DiathekeClient {
...
-    func newSession(model: String,
-                    success: @escaping (_ sessionID: String) -> (),
-                    failure: DiathekeFailureCallback?) {
-        var request = Cobaltspeech_Diatheke_NewSessionRequest()
-        request.model = model
-        
-        newSession(request).response.whenComplete { (result) in
+    // CreateSession creates a new Diatheke session using the specified model.
+    func createSession(modelID: String,
+                       success: @escaping (Cobaltspeech_Diatheke_SessionOutput) -> (),
```
Comment (jhollowayj) [inline grpc/swift-diatheke/Client.swift:97]: success is a function that we will call, passing in the session output, but what does the `@escaping` keyword mean?

**Review by jhollowayj — APPROVED** [review]
Seems fine to me.

File: grpc/swift-diatheke/Client.swift (hunk 1 above)
Comment (kapitoshka438) [reply grpc/swift-diatheke/Client.swift:97]: > success is a function that we will call, passing in the session output, but what does the `@escaping` keyword mean?

`@escaping` says that `success` function will be called outside of the `createSession` function.

File: grpc/swift-diatheke/diatheke.grpc.swift [hunk 2]
```diff
@@ -20,236 +20,199 @@
 // See the License for the specific language governing permissions and
```
[the remark this answers is not in the document]
Comment (kapitoshka438) [reply grpc/swift-diatheke/diatheke.grpc.swift:20]: > Should we be committing this file? Does the swift package manager pull from this generated file, or is there a good way to generate on _their_ computer as part of _their_ project it when they add the dependency?
> I just worry about forgetting to update this file and bump the version and do the release when I update the protobuf definition. This problem apply to more than just swift, so I'm just trying to get a little more information on what swift can do.

I need to make some research. This file was generated by swift protobuf plugin from `proto` file and it cantains all needed grpc types. If we exclude it from the package source code, it should be generated on client side when a developer adds SPM dependency to our SDK. And the swift protobuf plugin must be downloaded to the client, and some script should be added to `Package.swift`. And I even don't know if it is possible. And source files located in the package are non-editable, that is great. And I don't know if will work for on-client-generated files.

File: grpc/swift-diatheke/Diatheke.swift [hunk 3]
```diff
@@ -12,4 +12,4 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-public let DiathekeVersion = "1.0"
+public let DiathekeSDKVersion = "2.0"
```
[the remark this answers is not in the document]
Comment (kapitoshka438) [reply grpc/swift-diatheke/Diatheke.swift:15]: I agree. I think this value should match the tag.

[the remark this answers is not in the document]
Comment (kapitoshka438) [reply grpc/swift-diatheke/Diatheke.swift:15]: Oh, I remember. This value was automatically updated by `sed`, described in the Readme.md. But I have changed the variable name. I will fix the PR.

File: README.md [hunk 4]
```diff
@@ -84,7 +84,7 @@ NEW_VERSION="1.0.1"
 git checkout master
 git checkout -b version-update-v$NEW_VERSION
 
-sed -i 's|DiathekeVersion = "[0-9.]*"|DiathekeVersion = "'$NEW_VERSION'"|g' grpc/swift-diatheke/Diatheke.swift
+sed -i 's|DiathekeSDKVersion = "[0-9.]*"|DiathekeSDKVersion = "'$NEW_VERSION'"|g' grpc/swift-diatheke/Diatheke.swift
```
Comment (kapitoshka438) [inline README.md:87]: @jhollowayj this is where the `DiathekeSDKVersion` changes when we tag a new version

**Comment by mplooster** [comment]
This has been approved. Is there anything left preventing it from being merged? @kapitoshka438 @jhollowayj

**Pull request merged**